diff --git a/techpack/audio/asoc/msm8952-slimbus.c b/techpack/audio/asoc/msm8952-slimbus.c
--- a/techpack/audio/asoc/msm8952-slimbus.c
+++ b/techpack/audio/asoc/msm8952-slimbus.c
@@ -3792,9 +3792,10 @@ static int msm8952_mclk_event(struct snd_soc_dapm_widget *w,
 		struct snd_kcontrol *kcontrol, int event)
 {
-	struct snd_soc_codec *codec = snd_soc_dapm_to_codec(w->dapm);
 	pr_debug("%s: event = %d\n", __func__, event);
 
 #ifndef CONFIG_SND_SOC_MADERA
+	struct snd_soc_codec *codec = snd_soc_dapm_to_codec(w->dapm);
+
 	switch (event) {
 	case SND_SOC_DAPM_PRE_PMU:
 		return msm8952_enable_codec_mclk(codec, 1, true);
